casual/casual-network/src/main/java/se/laz/casual/network/outbound/JEEConcurrencyFactory.java

```diff
@@ -62,4 +70,48 @@ public static ManagedExecutorService getManagedExecutorService()
         }
     }
 
+    /**
+     * Gets the default ManagedScheduledExecutorService on jboss. On some non-standard implementation
+     * it will fetch the default through java:comp/DefaultManagedScheduledExecutorService
+     * which normally shouldn't be accessible in this context.
+     * <p>
+     * If both of those fail a ScheduledExecutorService instance will be created
+     * through the standard java.util.concurrent.Executor which will share the
+     * same ScheduledExecutorService among all users.
+     *
+     * @return A ScheduledExecutorService instance.
+     */
+    public static ScheduledExecutorService getManagedScheduledExecutorService()
+    {
+        try
+        {
+            // First try jboss variant
+            return InitialContext.doLookup(DEFAULT_MANAGED_SCHEDULED_EXECUTOR_SERVICE_NAME_JBOSS_DIRECT);
+        }
+        catch (NamingException e)
+        {
+            LOG.info("Failed lookup for " + DEFAULT_MANAGED_SCHEDULED_EXECUTOR_SERVICE_NAME_JBOSS_DIRECT + ", will retry indirect jndi name (may exist in this context on some non-standard application servers)");
+            try
+            {
+                // Second try non-standard use of indirect jndi name defined in JSR-236
+                return InitialContext.doLookup(DEFAULT_MANAGED_SCHEDULED_EXECUTOR_SERVICE_NAME_INDIRECT);
+            }
+            catch (NamingException ex)
+            {
+                LOG.info("Failed lookup for " + DEFAULT_MANAGED_SCHEDULED_EXECUTOR_SERVICE_NAME_INDIRECT + ", will use scheduled executor from java.util.concurrent.Executors");
+                // If all else fails, use java.util.concurrent variant as scheduler
+                return getSharedJavaUtilScheduledExecutor();
+            }
+        }
+    }
+
+    private static synchronized ScheduledExecutorService getSharedJavaUtilScheduledExecutor()
+    {
+        if (null == sharedScheduledExecutor)
+        {
+            sharedScheduledExecutor = Executors.newScheduledThreadPool(10);
```

Comment:
10, because? Just curious about sizing what this impacts, would it need to be higher than the number of reverse connections or? Which then leads me to wonder, what is the configuration of the weblogic / wildfly versions, do they need to be configured similarly. Just would be good to know from a documentation perspective, in which case we might want to make this 10 configurable too.

Reply:
Basically arbitrarily chosen. I felt like it needed to be able to handle a bunch of concurrent calls since it's a shared scheduler, but wanted to be able to push something before the weekend. I agree it should be configurable, but does it need to be added right now? This particular scheduler won't be used in a production case at the moment. If it isn't made configurable now an issue should be created to add config later, though

Reply:
If you add quick boolean env lookup, then we can fix it properly in the other issue related to getting everything inside ConfigurationService: https://github.com/casualcore/casual-java/issues/44